Tasks/PublishPipelineArtifactV1/task.json

```diff
@@ -38,6 +38,48 @@
             "defaultValue": "",
             "required": false,
             "helpMarkDown": "Name of the artifact to publish. If not set, defaults to a unique ID scoped to the job."
+        }, 
+        {
+            "name": "artifactType",
+            "aliases": [
+                "publishLocation"
+            ],
+            "type": "pickList",
+            "label": "Artifact publish location",
+            "defaultValue": "pipeline",
+            "required": true,
+            "helpMarkDown": "Choose whether to store the artifact in Azure Pipelines, or to copy it to a file share that must be accessible from the pipeline agent.",
+            "options": {
+                "pipeline": "Azure Pipelines",
+                "filepath": "A file share"
+            }
+        },
+        {
+            "name": "fileSharePath",
+            "type": "string",
+            "label": "File share path",
+            "defaultValue": "",
+            "required": true,
+            "helpMarkDown": "The file share to which the artifact files will be copied. This can include variables. Example: \\\\\\\\my\\\\share\\\\$(Build.DefinitionName)\\\\$(Build.BuildNumber). Publishing artifacts from a Linux or macOS agent to a file share is not supported.",
+            "visibleRule": "artifactType = filepath"
+        },
+        {
+            "name": "parallel",
+            "type": "boolean",
+            "label": "Parallel copy",
+            "defaultValue": "false",
+            "required": false,
+            "helpMarkDown": "Select whether to copy files in parallel using multiple threads for greater potential throughput. If this setting is not enabled, one thread will be used.",
+            "visibleRule": "artifactType = filepath"
+        },
+        {
+            "name": "parallelCount",
```

Comment:
The parallelism is only for FileShare right? Or are we going to use for both - PA/UNC ?

Reply:
Not in the scope of this PR I think.